### ISSUE TITLE
Admin server status check does not work with two similar named domains

### ISSUE THREAD

**Issue by JoshiiSinfield** [biemond/biemond-orawls#226]
Hi,

I have two domains configured on the same server with similar names : loadtest & loadtesting.

This is incompatible with the ps/grep command used in wls_adminserver.rb for the status check, as the grep for loadtest actually returns the loadtesting process!

please see the example debug output below!

Cheers,
Josh

Debug: adminserver_status /bin/ps -ef | /bin/grep -v grep | /bin/grep 'weblogic.Name=AdminServer' | /bin/grep loadtesting
Debug: line wls      14135 13999  0 17:35 ?        00:00:42 /usr/java/jdk1.7.0_79/jre/bin/java -Dweblogic.Name=AdminServer -Dbea.home=/opt/oracle/middleware -Djava.security.policy=/opt/oracle/middleware/wlserver_10.3/server/lib/weblogic.policy -Djava.library.path=/opt/oracle/middleware/patch_wls1036/profiles/default/native:/opt/oracle/middleware/patch_ocp371/profiles/default/native:/opt/oracle/middleware/wlserver_10.3/server/native/linux/x86_64:/opt/oracle/middleware/wlserver_10.3/server/native/linux/x86_64/oci920_8:/usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib -Djava.class.path=/opt/oracle/middleware/patch_wls1036/profiles/default/sys_manifest_classpath/weblogic_patch.jar:/opt/oracle/middleware/patch_ocp371/profiles/default/sys_manifest_classpath/weblogic_patch.jar:/usr/java/latest/lib/tools.jar:/opt/oracle/middleware/wlserver_10.3/server/lib/weblogic_sp.jar:/opt/oracle/middleware/wlserver_10.3/server/lib/weblogic.jar:/opt/oracle/middleware/modules/features/weblogic.server.modules_10.3.6.0.jar:/opt/oracle/middleware/wlserver_10.3/server/lib/webservices.jar:/opt/oracle/middleware/modules/org.apache.ant_1.7.1/lib/ant-all.jar:/opt/oracle/middleware/modules/net.sf.antcontrib_1.1.0.0_1-0b2/lib/ant-contrib.jar::/opt/oracle/middleware -Dweblogic.system.BootIdentityFile=/opt/oracle/wlsdomains/loadtesting/servers/AdminServer/security/boot.properties -Dweblogic.nodemanager.ServiceEnabled=true -XX:PermSize=256m -XX:MaxPermSize=512m -Xms1024m -Xmx1024m -Dweblogic.rjvm.enableprotocolswitch=true weblogic.Server

Debug: found server
Debug: adminserver_status output Found
Debug: adminserver_status /bin/ps -ef | /bin/grep -v grep | /bin/grep 'weblogic.Name=AdminServer' | /bin/grep loadtest
Debug: line wls      14135 13999  0 17:35 ?        00:00:42 /usr/java/jdk1.7.0_79/jre/bin/java -Dweblogic.Name=AdminServer -Dbea.home=/opt/oracle/middleware -Djava.security.policy=/opt/oracle/middleware/wlserver_10.3/server/lib/weblogic.policy -Djava.library.path=/opt/oracle/middleware/patch_wls1036/profiles/default/native:/opt/oracle/middleware/patch_ocp371/profiles/default/native:/opt/oracle/middleware/wlserver_10.3/server/native/linux/x86_64:/opt/oracle/middleware/wlserver_10.3/server/native/linux/x86_64/oci920_8:/usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib -Djava.class.path=/opt/oracle/middleware/patch_wls1036/profiles/default/sys_manifest_classpath/weblogic_patch.jar:/opt/oracle/middleware/patch_ocp371/profiles/default/sys_manifest_classpath/weblogic_patch.jar:/usr/java/latest/lib/tools.jar:/opt/oracle/middleware/wlserver_10.3/server/lib/weblogic_sp.jar:/opt/oracle/middleware/wlserver_10.3/server/lib/weblogic.jar:/opt/oracle/middleware/modules/features/weblogic.server.modules_10.3.6.0.jar:/opt/oracle/middleware/wlserver_10.3/server/lib/webservices.jar:/opt/oracle/middleware/modules/org.apache.ant_1.7.1/lib/ant-all.jar:/opt/oracle/middleware/modules/net.sf.antcontrib_1.1.0.0_1-0b2/lib/ant-contrib.jar::/opt/oracle/middleware -Dweblogic.system.BootIdentityFile=/opt/oracle/wlsdomains/loadtesting/servers/AdminServer/security/boot.properties -Dweblogic.nodemanager.ServiceEnabled=true -XX:PermSize=256m -XX:MaxPermSize=512m -Xms1024m -Xmx1024m -Dweblogic.rjvm.enableprotocolswitch=true weblogic.Server

**Comment by biemond** [comment]
Hi,

I am aware of this problem, the alternative is startup wlst , connect to the nodemanager and check for the status but this way is very slow.
Now I am only  doing this for cluster.

Maybe I should make a switch on this type so you can also check it on the slow way 

Cheers

**Comment by JoshiiSinfield** [comment]
Hi Edwin,

I imagine that would be slow yes. Just from looking at the code, I see that domain_path is available. I haven't tested it yet, but can we do something like this:
command  = "#{ps_bin} #{ps_arg} | /bin/grep -v grep | /bin/grep '#{domain_path}/servers/#{name}'

to create this command:
 /bin/ps -ef | /bin/grep -v grep | /bin/grep '/opt/oracle/wlsdomains/loadtesting/servers/AdminServer'

which does the following:
[root@~]# /bin/ps -ef | /bin/grep -v grep | /bin/grep '/opt/oracle/wlsdomains/loadtesting/servers/AdminServer'
wls      14135 13999  0 Sep22 ?        00:01:36 /usr/java/jdk1.7.0_79/jre/bin/java -Dweblogic.Name=AdminServer -Dbea.home=/opt/oracle/middleware -Djava.security.policy=/opt/oracle/middleware/wlserver_10.3/server/lib/weblogic.policy -Djava.library.path=/opt/oracle/middleware/patch_wls1036/profiles/default/native:/opt/oracle/middleware/patch_ocp371/profiles/default/native:/opt/oracle/middleware/wlserver_10.3/server/native/linux/x86_64:/opt/oracle/middleware/wlserver_10.3/server/native/linux/x86_64/oci920_8:/usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib -Djava.class.path=/opt/oracle/middleware/patch_wls1036/profiles/default/sys_manifest_classpath/weblogic_patch.jar:/opt/oracle/middleware/patch_ocp371/profiles/default/sys_manifest_classpath/weblogic_patch.jar:/usr/java/latest/lib/tools.jar:/opt/oracle/middleware/wlserver_10.3/server/lib/weblogic_sp.jar:/opt/oracle/middleware/wlserver_10.3/server/lib/weblogic.jar:/opt/oracle/middleware/modules/features/weblogic.server.modules_10.3.6.0.jar:/opt/oracle/middleware/wlserver_10.3/server/lib/webservices.jar:/opt/oracle/middleware/modules/org.apache.ant_1.7.1/lib/ant-all.jar:/opt/oracle/middleware/modules/net.sf.antcontrib_1.1.0.0_1-0b2/lib/ant-contrib.jar::/opt/oracle/middleware -Dweblogic.system.BootIdentityFile=/opt/oracle/wlsdomains/loadtesting/servers/AdminServer/security/boot.properties -Dweblogic.nodemanager.ServiceEnabled=true -XX:PermSize=256m -XX:MaxPermSize=512m -Xms1024m -Xmx1024m -Dweblogic.rjvm.enableprotocolswitch=true weblogic.Server
[root@~]# /bin/ps -ef | /bin/grep -v grep | /bin/grep '/opt/oracle/wlsdomains/loadtest/servers/AdminServer'
[root@~]#



or even a command like this:
command  = "#{ps_bin} #{ps_arg} | /bin/grep -v grep | /bin/grep '#{domain_name}/servers/#{name}'

Cheers

**Comment by biemond** [comment]
this will check BootIdentityFile property but I know this property is not always set, is optional or used.

**Comment by JoshiiSinfield** [comment]
indeed, but on that premise isn't the current check also flawed in the same way - as the only place the domain name appears currently is in the BootIdentityFile property?

The only other thing I can think of doing is interrogating the AdminServer.pid file at /opt/oracle/wlsdomains/loadtesting/servers/AdminServer/data/nodemanager/AdminServer.pid, and perhaps checking if the AdminServer.lck and /opt/oracle/wlsdomains/loadtesting/servers/AdminServer/tmp/AdminServer.lok files are present then reading in the pid from the AdminServer.pid file and doing a ps -ef | grep -v grep | grep '#{existing_pid}'

**Comment by biemond** [comment]
yeah, that is the best and fastest way.

and the orawls module requires a nodemanager so this pid file should always be there.
